In [29]:
import pandas as pd

# import data GlobalWeatherRepository
path = "GlobalWeatherRepository.csv"
df = pd.read_csv(path)
df

,country,location_name,latitude,longitude,timezone,last_updated_epoch,last_updated,temperature_celsius,temperature_fahrenheit,condition_text,...,air_quality_PM2.5,air_quality_PM10,air_quality_us-epa-index,air_quality_gb-defra-index,sunrise,sunset,moonrise,moonset,moon_phase,moon_illumination
0,Afghanistan,Kabul,34.5200,69.1800,Asia/Kabul,1715849100,5/16/2024 13:15,26.6,79.8,Partly Cloudy,...,8.400,26.600,1,1,4:50 AM,6:50 PM,12:12 PM,1:11 AM,Waxing Gibbous,55
1,Albania,Tirana,41.3300,19.8200,Europe/Tirane,1715849100,5/16/2024 10:45,19.0,66.2,Partly cloudy,...,1.100,2.000,1,1,5:21 AM,7:54 PM,12:58 PM,2:14 AM,Waxing Gibbous,55
2,Algeria,Algiers,36.7600,3.0500,Africa/Algiers,1715849100,5/16/2024 9:45,23.0,73.4,Sunny,...,10.400,18.400,1,1,5:40 AM,7:50 PM,1:15 PM,2:14 AM,Waxing Gibbous,55
3,Andorra,Andorra La Vella,42.5000,1.5200,Europe/Andorra,1715849100,5/16/2024 10:45,6.3,43.3,Light drizzle,...,0.700,0.900,1,1,6:31 AM,9:11 PM,2:12 PM,3:31 AM,Waxing Gibbous,55
4,Angola,Luanda,-8.8400,13.2300,Africa/Luanda,1715849100,5/16/2024 9:45,26.0,78.8,Partly cloudy,...,183.400,262.300,5,10,6:12 AM,5:55 PM,1:17 PM,12:38 AM,Waxing Gibbous,55
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
39589,Venezuela,Caracas,10.5000,-66.9167,America/Caracas,1733482800,12/6/2024 7:00,16.2,61.2,Sunny,...,14.430,19.240,1,2,6:33 AM,6:05 PM,11:11 AM,11:08 PM,Waxing Crescent,24
39590,Vietnam,Hanoi,21.0333,105.8500,Asia/Bangkok,1733481000,12/6/2024 17:30,21.2,70.2,Partly cloudy,...,99.160,100.825,4,10,6:20 AM,5:15 PM,10:35 AM,9:55 PM,Waxing Crescent,20
39591,Yemen,Sanaa,15.3547,44.2067,Asia/Aden,1733481900,12/6/2024 13:45,22.2,71.9,Sunny,...,10.360,31.080,1,1,6:16 AM,5:32 PM,10:39 AM,10:19 PM,Waxing Crescent,21
39592,Zambia,Lusaka,-15.4167,28.2833,Africa/Lusaka,1733481900,12/6/2024 12:45,36.0,96.8,Sunny,...,14.245,14.245,1,2,5:28 AM,6:28 PM,9:58 AM,11:06 PM,Waxing Crescent,22


# Tugas Nomor 1
## Analisa Korelasi: temperature_fahrenheit, pressure_mb, dan air_quality_Sulphur_dioxide

Data berhasil diimport dari GlobalWeatherRepository.csv, berisi 39594 baris dan 41 kolom.

In [22]:
kolom_analisis = df[['temperature_fahrenheit', 'pressure_mb', 'air_quality_Sulphur_dioxide']]
kolom_analisis

,temperature_fahrenheit,pressure_mb,air_quality_Sulphur_dioxide
0,79.8,1012,0.200
1,66.2,1012,0.100
2,73.4,1011,13.400
3,43.3,1007,0.200
4,78.8,1011,31.500
...,...,...,...
39589,61.2,1015,2.775
39590,70.2,1013,68.265
39591,71.9,1016,2.035
39592,96.8,1007,11.470


### Langkah 1: Mengambil kolom yang akan dianalisis
Dari keseluruhan 41 kolom di dataset, hanya 3 kolom berikut yang relevan untuk analisis ini: `temperature_fahrenheit`, `pressure_mb`, dan `air_quality_Sulphur_dioxide`.

In [23]:
print(kolom_analisis.describe())

       temperature_fahrenheit   pressure_mb  air_quality_Sulphur_dioxide
count            39594.000000  39594.000000                 39594.000000
mean                75.832651   1013.195914                     8.974035
std                 14.634566      6.689236                    54.608971
min                 -7.400000    971.000000                 -9999.000000
25%                 68.400000   1010.000000                     0.555000
50%                 79.000000   1013.000000                     1.850000
75%                 84.600000   1016.000000                     6.845000
max                120.600000   1080.000000                   400.340000


## Analisa statistik deskriptif

temperature_fahrenheit
Rata-rata suhu di seluruh data adalah 75.83 derajat fahrenheit, dengan standar deviasi 14.63, artinya suhu di data ini cukup bervariasi antar lokasi dan waktu. Suhu terendah tercatat -7.4 derajat dan tertinggi 120.6 derajat, jadi rentang datanya cukup lebar dari sangat dingin sampai sangat panas. Q1 nya 68.4, Q2 atau median 79.0, dan Q3 nya 84.6. Karena median lebih dekat ke Q3, sebaran data sedikit condong ke suhu yang lebih hangat, artinya lebih banyak data dengan suhu hangat atau panas dibanding suhu dingin.

pressure_mb
Rata-rata tekanan udara 1013.19 mb, dengan standar deviasi kecil yaitu 6.69, artinya tekanan udara relatif stabil dan tidak terlalu bervariasi dibanding suhu. Rentang datanya dari 971 mb sampai 1080 mb, cukup wajar untuk data tekanan atmosfer permukaan bumi. Q1 nya 1010, Q2 nya 1013, Q3 nya 1016. Karena ketiganya berdekatan dan dekat dengan mean, sebaran data tekanan udara cenderung simetris, tidak condong ke satu sisi.

air_quality_Sulphur_dioxide
Rata-rata terlihat 8.97, tapi standar deviasinya sangat besar yaitu 54.61 dibanding rata-ratanya sendiri. Ini menandakan ada data yang sangat ekstrem yang menarik nilai statistiknya menjadi tidak wajar. Nilai minimum tercatat -9999, yang jelas bukan angka polusi udara asli karena kadar polutan tidak mungkin negatif sebesar itu. Setelah dicek, ternyata ini adalah 1 baris data dari negara Kiribati yang kemungkinan besar merupakan kode data hilang atau error dari sumber datanya, bukan pengukuran sungguhan. Karena hanya 1 dari 39594 baris atau sekitar 0.0025 persen dari total data, nilai ini tidak dibuang dari dataset utuh tapi cukup diwaspadai saat membaca angka mean dan std, karena kedua angka itu jadi kurang mewakili kondisi data yang sebenarnya. Melihat Q1 nya 0.555, Q2 nya 1.85, dan Q3 nya 6.845, sebagian besar data kadar SO2 sebenarnya berada di rentang yang jauh lebih masuk akal dan rendah, jauh berbeda dari kesan yang diberikan oleh nilai mean atau std di atas.

In [24]:
korelasi = kolom_analisis.corr()
print(korelasi)

                             temperature_fahrenheit  pressure_mb  \
temperature_fahrenheit                     1.000000    -0.486833   
pressure_mb                               -0.486833     1.000000   
air_quality_Sulphur_dioxide               -0.023152     0.018620   

                             air_quality_Sulphur_dioxide  
temperature_fahrenheit                         -0.023152  
pressure_mb                                     0.018620  
air_quality_Sulphur_dioxide                     1.000000  


### Analisa Korelasi

1. temperature_fahrenheit vs pressure_mb = -0.486833
- Ini adalah korelasi negatif dengan kekuatan sedang menuju kuat (mendekati -0.5 dari skala -1 sampai 1).
- Artinya: semakin tinggi suhu udara, semakin rendah tekanan udaranya, dan berlaku sebaliknya juga (semakin dingin suhunya, semakin tinggi tekanan udaranya).
- Hubungan ini masuk akal secara fisika: udara yang panas memiliki kepadatan molekul yang lebih rendah (molekul udara bergerak lebih cepat dan menyebar), sehingga tekanan yang dihasilkan terhadap permukaan menjadi lebih kecil. Sebaliknya udara dingin lebih padat sehingga tekanannya lebih tinggi.
- Karena nilainya cukup jauh dari 0 (mendekati -0.5), hubungan ini bisa dibilang cukup nyata dan konsisten di dalam dataset ini, bukan sekadar kebetulan.

2. temperature_fahrenheit vs air_quality_Sulphur_dioxide = -0.023152
- Nilai ini sangat mendekati 0, sehingga bisa disimpulkan tidak ada hubungan linear yang berarti antara suhu udara dan kadar Sulphur Dioxide di udara.
- Meskipun tandanya negatif (artinya arahnya sedikit berlawanan), kekuatannya terlalu lemah untuk dianggap sebagai pola yang nyata. Nilai sekecil ini bisa saja muncul murni karena variasi acak dalam data, bukan hubungan sebab-akibat.
- Kesimpulannya: naik-turunnya suhu tidak bisa dipakai untuk memprediksi naik-turunnya kadar SO2 di suatu lokasi.

3. pressure_mb vs air_quality_Sulphur_dioxide = 0.018620
- Sama seperti poin sebelumnya, nilai ini sangat mendekati 0, menunjukkan tidak ada hubungan linear yang berarti antara tekanan udara dan kadar Sulphur Dioxide.
- Tandanya positif, tapi karena kekuatannya sangat lemah (jauh di bawah 0.1), ini tidak bisa dianggap sebagai indikasi hubungan yang nyata antar kedua variabel.

### Kesimpulan Akhir

Dari ketiga pasangan variabel yang dianalisis, hanya temperature_fahrenheit dan pressure_mb yang menunjukkan hubungan yang cukup jelas dan bisa dijelaskan secara ilmiah (berbanding terbalik, sesuai prinsip fisika atmosfer). Sementara itu, air_quality_Sulphur_dioxide ternyata tidak dipengaruhi secara berarti baik oleh suhu maupun tekanan udara.

Ini masuk akal karena kadar Sulphur Dioxide di udara umumnya lebih dipengaruhi oleh sumber polusi lokal seperti asap kendaraan bermotor, aktivitas pabrik/industri, pembakaran bahan bakar fosil, atau letak geografis suatu wilayah (misalnya dekat kawasan industri atau perkotaan padat), bukan oleh kondisi cuaca seperti suhu dan tekanan udara semata. Dengan kata lain, untuk memprediksi kadar SO2 di suatu tempat, data cuaca saja tidak cukup, dibutuhkan data tambahan seperti tingkat aktivitas industri atau kepadatan lalu lintas di wilayah tersebut.

In [25]:
import pandas as pd

path = "shopping_trends_updated.csv"
df = pd.read_csv(path)
df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,Express,No,No,24,Venmo,Weekly


# Tugas Nomor 2
## Analisa Dataset Customer Shopping Trends

Dataset yang digunakan diambil dari Kaggle: [Customer Shopping Trends Dataset](https://www.kaggle.com/datasets/iamsouravbanerjee/customer-shopping-trends-dataset)

Data berhasil diimport dari **shopping_trends_updated.csv**, berisi 3900 baris dan 18 kolom, tanpa ada data yang kosong (missing value).

In [26]:
print(df['Previous Purchases'].describe())

count    3900.000000
mean       25.351538
std        14.447125
min         1.000000
25%        13.000000
50%        25.000000
75%        38.000000
max        50.000000
Name: Previous Purchases, dtype: float64


### Analisa Statistik Deskriptif: Previous Purchases

Hasil yang keluar:

| Statistik | Nilai |
|---|---|
| count | 3900 |
| mean | 25.351538 |
| std | 14.447125 |
| min | 1.0 |
| 25% (Q1) | 13.0 |
| 50% (Q2 / median) | 25.0 |
| 75% (Q3) | 38.0 |
| max | 50.0 |

Penjelasan detail:

- Mean (25.35): rata-rata, setiap pelanggan dalam dataset ini sudah pernah melakukan pembelian sebanyak 25 kali sebelumnya. Ini menunjukkan basis pelanggan yang dianalisis termasuk pelanggan yang cukup aktif/loyal secara keseluruhan.

- Std (14.45): standar deviasi ini termasuk besar dibandingkan rata-ratanya sendiri (lebih dari setengah nilai mean). Ini berarti jumlah pembelian sebelumnya antar pelanggan sangat bervariasi, ada pelanggan yang benar-benar baru, ada juga yang sudah sangat sering belanja.

- Min (1) dan Max (50): rentang datanya terbentang dari pelanggan yang baru pertama kali tercatat (1 kali pembelian) sampai pelanggan paling loyal (50 kali pembelian). Rentang sejauh ini (49 poin) menunjukkan populasi pelanggan yang beragam, dari yang baru sampai yang sangat setia.

- Q1 (13): 25% pelanggan dengan riwayat pembelian paling sedikit, jumlah pembelian sebelumnya mereka ada di bawah 13 kali. Ini adalah kelompok pelanggan yang relatif baru.

- Q2 / median (25): setengah dari seluruh pelanggan memiliki riwayat pembelian di bawah 25 kali, dan setengah lagi di atas 25 kali. Titik tengah data.

- Q3 (38): 75% pelanggan memiliki riwayat pembelian di bawah 38 kali, sehingga 25% sisanya (pelanggan paling loyal) sudah membeli lebih dari 38 kali.

- Kesimpulan sebaran data: karena nilai median (25) sangat dekat dengan mean (25.35), dan jarak Q1 ke Q2 (13 ke 25 = 12 poin) hampir sama dengan jarak Q2 ke Q3 (25 ke 38 = 13 poin), data ini tersebar secara cukup merata dan simetris, tidak menumpuk di satu sisi (tidak skewed). Artinya jumlah pelanggan baru, menengah, dan sangat loyal proporsinya relatif seimbang dalam dataset ini.

In [27]:
korelasi_age = df[['Age', 'Purchase Amount (USD)']].corr()
print(korelasi_age)

                            Age  Purchase Amount (USD)
Age                    1.000000              -0.010424
Purchase Amount (USD) -0.010424               1.000000


### Analisa Korelasi: Age vs Purchase Amount (USD)

Hasil korelasi: -0.010424

Penjelasan detail:

- Nilai korelasi ini sangat mendekati 0 (skala korelasi berkisar dari -1 sampai 1), sehingga termasuk kategori korelasi yang dapat diabaikan / tidak berarti secara praktis.
- Tandanya negatif, tapi karena besarnya kurang dari 0.1, tanda ini tidak punya makna praktis, bisa jadi hanya fluktuasi acak dalam data, bukan pola sungguhan.
- Artinya: usia pelanggan tidak bisa dipakai untuk memprediksi seberapa besar nominal uang yang mereka belanjakan. Pelanggan muda dan pelanggan tua memiliki kecenderungan pola belanja (dalam nominal USD) yang relatif sama, tidak ada kelompok usia tertentu yang secara konsisten belanja lebih besar atau lebih kecil.
- Ini masuk akal jika dataset ini bersifat sintetis/acak (seperti disebutkan di deskripsi Kaggle-nya sebagai "Synthetic Data"), di mana nominal pembelian memang sengaja tidak dikaitkan dengan usia pelanggan.

In [28]:
korelasi_rating = df[['Purchase Amount (USD)', 'Review Rating']].corr()
print(korelasi_rating)

                       Purchase Amount (USD)  Review Rating
Purchase Amount (USD)               1.000000       0.030776
Review Rating                       0.030776       1.000000


### Analisa Korelasi: Purchase Amount (USD) vs Review Rating

Hasil korelasi: 0.030776

Penjelasan detail:

- Sama seperti korelasi sebelumnya, nilai ini sangat mendekati 0, sehingga termasuk kategori korelasi yang dapat diabaikan.
- Kali ini tandanya positif, tapi karena nilainya jauh di bawah 0.1, ini tidak cukup kuat untuk disebut sebagai hubungan yang nyata antar dua variabel.
- Artinya: besar-kecilnya nominal uang yang dibelanjakan pelanggan tidak berhubungan dengan rating ulasan (review rating) yang mereka berikan. 
  - Pelanggan yang belanja dalam jumlah besar belum tentu memberikan rating yang tinggi.
  - Pelanggan yang memberikan rating tinggi belum tentu juga adalah pelanggan yang belanja dalam jumlah besar.
- Ini menunjukkan bahwa kepuasan pelanggan (diwakili oleh review rating) kemungkinan lebih dipengaruhi oleh faktor lain seperti kualitas produk, pelayanan, atau ekspektasi pribadi, bukan oleh seberapa mahal barang yang mereka beli.

### Kesimpulan Akhir Tugas Nomor 2

Dari kedua analisis korelasi di atas, dapat disimpulkan bahwa nominal pembelian (Purchase Amount) pada dataset ini tidak dipengaruhi secara berarti oleh usia pelanggan maupun rating ulasan yang diberikan. Kedua korelasi menunjukkan angka yang sangat mendekati nol, yang berarti tidak ada hubungan linear yang bisa diandalkan antar variabel-variabel tersebut. 

Jika bisnis ingin memahami faktor apa yang benar-benar mempengaruhi besar-kecilnya nominal belanja pelanggan, perlu dilakukan analisis terhadap variabel lain di luar usia dan rating, misalnya kategori produk yang dibeli, musim (season), metode pembayaran, atau status subscription pelanggan.